src/Compilers/Core/Portable/SourceGeneration/Nodes/NodeStateTable.cs

```diff
@@ -88,6 +88,14 @@ private NodeStateTable(ImmutableArray<TableEntry> states, ImmutableArray<Increme
 
         public ImmutableArray<IncrementalGeneratorRunStep> Steps { get; }
 
+        public int GetTotalEntryItemCount()
+        {
+            var count = 0;
+            foreach (var inputEntry in _states)
+                count += inputEntry.Count;
+            return count;
```

Comment:
```suggestion
    return _states.Aggregate(0, (count, i) => count + i);
```

Reply:
Or add a `Sum` to our `ImmutableArrayExtensions` that takes a functor.

Reply:
i'm totally inconsistent here.  i do Linq+StaticLambdas in some places, but foreaches in others :)